### PR TITLE
Add between method in `select.rb` and tests.

### PR DESCRIPTION
**Pull request by komainu8**

### CODE REVIEW

File: lib/groonga/client/request/select.rb [hunk 1]
```diff
@@ -97,6 +97,16 @@ def query(value)
         #     The new request object for setting a filter condition.
         #
         #   @since 0.4.3
+        #
+        #   @example: Use between function
+        #      request.
+        #        filter.between("tags", "min", "min_border", "max", "max_border")
+        #          # -> --filter 'between(tags, min, "min_border", max, "max_border")'
```
Comment (kou) [inline lib/groonga/client/request/select.rb]: Example should be "workable". `"min_border"` and `"max_border"` are "invalid" border. It should be `"include"` or `"exclude"`. And `"min"` and `"max"` should be more useful values.

File: lib/groonga/client/request/select.rb [hunk 2]
```diff
@@ -97,6 +97,16 @@ def query(value)
         #     The new request object for setting a filter condition.
         #
         #   @since 0.4.3
+        #
+        #   @example: Use between function
```
Comment (kou) [inline lib/groonga/client/request/select.rb]: Please move to the below of the first `@example`. `@example`s should be close.

File: lib/groonga/client/request/select.rb [hunk 3]
```diff
@@ -97,6 +97,16 @@ def query(value)
...
+        #      request.
+        #        filter.between("tags", "min", "min_border", "max", "max_border")
+        #          # -> --filter 'between(tags, min, "min_border", max, "max_border")'
+        #
+        #   @return [Groonga::Client::Request::Select::Filter]
+        #     The new request object for setting a filter condition.
```
Comment (kou) [inline lib/groonga/client/request/select.rb]: It's duplicated.

File: lib/groonga/client/request/select.rb [hunk 4]
```diff
@@ -97,6 +97,16 @@ def query(value)
...
+        #
+        #   @since 0.4.4
```
Comment (kou) [inline lib/groonga/client/request/select.rb]: `filter` can be used `@since 0.4.3`. It's needless.

File: lib/groonga/client/request/select.rb [hunk 5]
```diff
@@ -182,6 +192,31 @@ def initialize(request)
             @request = request
           end
 
+          # Adds a `between` condition then return a new `select`
+          # request object.
+          #
+          # @example: Multiple conditions
+          #    request.
+          #      filter.between("tags", "min", "min_border", "max", "max_border").
+          #        # -> --filter 'between(tags, min, "min_border", max, "max_border")'
```
Comment (kou) [inline lib/groonga/client/request/select.rb]: It's not a "multiple conditions" case.

File: lib/groonga/client/request/select.rb [hunk 6]
```diff
@@ -182,6 +192,31 @@ def initialize(request)
...
+          #
+          # @example: Ignore no values case
+          #    request.
+          #      filter.between("tags")
+          #        # -> --filter ''
```
Comment (kou) [inline lib/groonga/client/request/select.rb]: It should be treated as an error.

File: lib/groonga/client/request/select.rb [hunk 7]
```diff
@@ -182,6 +192,31 @@ def initialize(request)
...
+          #
+          # @param column_name_or_value [String, Symbol, Integer] The target column name
+          #   or target num.
```
Comment (kou) [inline lib/groonga/client/request/select.rb]: It should be `'column_name`. `value` is meaningless for `between`.

File: lib/groonga/client/request/select.rb [hunk 8]
```diff
@@ -182,6 +192,31 @@ def initialize(request)
...
+          #
+          # @param values [Object] Range of values.
+          #
+          # @return [Groonga::Client::Request::Select]
+          #   The new request with the given condition.
+          def between(column_name_or_value, *values)
```
Comment (kou) [inline lib/groonga/client/request/select.rb]: Please don't use needless variable length arguments. If we can use fixed length arguments, we should use it for easy to use and easy to maintain.

**Comment by komainu8** [comment]
@kou Thank you very much for your comments. I've fixed some documents and API arguments.

**Comment by kou** [comment]
Merged.
I cleaned your works after I merged.

**Pull request merged**